Frameworks/ImageIO/CGImageDestination.mm

```diff
@@ -15,11 +16,867 @@
+            _type = typeBMP;
+            RETURN_NULL_IF_FAILED(_idFactory->CreateEncoder(GUID_ContainerFormatBmp, NULL, &_idEncoder));
+        } else {
+            _type = typeUnknown;
+            return NULL;
+        }
+
+        if (url && [(NSURL*)url path]) { // Make sure that the url path actually resolves to something to avoid causing an exception
+            NSString* urlNSString = [[(NSURL*)url path] substringFromIndex:1];
+            NSData* urlAsData = [urlNSString dataUsingEncoding:NSUnicodeStringEncoding];
+            std::wstring wideUrl((wchar_t*)[urlAsData bytes], [urlAsData length]/sizeof(wchar_t));
+            RETURN_NULL_IF_FAILED(_idStream->InitializeFromFilename(wideUrl.c_str(), GENERIC_WRITE));
+
+            if (![(NSURL*)url checkResourceIsReachableAndReturnError:NULL]) {
+                return NULL;
+            }
+        } else if (data) {
+            _outData = data;
+
+            IStream* dataStream;
+            CreateStreamOnHGlobal(NULL, true, &dataStream);
+            RETURN_NULL_IF_FAILED(_idStream->InitializeFromIStream(dataStream));
+        } else {
+            // We are not currently handling data consumer as destination because data consumer is not implemented in WinObjC
+            UNIMPLEMENTED_WITH_MSG("Destination as Data Consumer is not handled right now");
+        }
+
+        RETURN_NULL_IF_FAILED(_idEncoder->Initialize(_idStream.Get(), WICBitmapEncoderNoCache));
+
+        if (_type == typeGIF) {
+            RETURN_NULL_IF_FAILED(_idEncoder->GetMetadataQueryWriter(&_idGifEncoderMetadataQueryWriter));
+        }
+    }
+
+    return self;
+}
+
+@end
+
+// Helper function for setting the property write values when formatting image metadata.
+// Image metadata stores decimal numbers as an integer divided by another integer.
+// The HighPart is a number representing what to divide the LowPart by to get the actual value.
+void setHighLowPartsUnsigned(ULARGE_INTEGER* valueLarge, double valueDouble) {
+    if (valueDouble < 0) {
+        NSTraceInfo(TAG, @"Tried to pass negative value to set unsigned long!\n");
+        return;
+    }
+
+    // Check to see if the value has a decimal component. If not, just divide by 1.
+    if (valueDouble == (unsigned int)valueDouble) {
+        (*valueLarge).LowPart = (unsigned int)valueDouble;
+        (*valueLarge).HighPart = 1;
+    } else {
+        (*valueLarge).LowPart = (unsigned int)(valueDouble * 100);
+        (*valueLarge).HighPart = 100;
+    }
+}
+
+// Signed version of the above
+void setHighLowPartsSigned(LARGE_INTEGER* valueLarge, double valueDouble) {
+    // Check to see if the value has a decimal component. If not, just divide by 1.
+    if (valueDouble == (int)valueDouble) {
+        (*valueLarge).LowPart = (int)valueDouble;
+        (*valueLarge).HighPart = 1;
+    } else {
+        (*valueLarge).LowPart = (int)(valueDouble * 100);
+        (*valueLarge).HighPart = 100;
+    }
+}
```

Comment:
nit we could condense these into a function template;

template <typename TOut>
void setHighLowParts(TOut& out, double value) {
    ...
}